### PR TITLE
Add option to allow nodal-to-elemental coupling

### LINKED ISSUE #21434: Allow nodal to elemental variable coupling
## Reason
`Coupleable` does not currently allow coupling an elemental variable when the primary variable is nodal. There exists circumstances when elemental integral values can be formed by nodal values.

## Design
Add an optional private parameter to 'Coupleable' that will skip nodal-to-element coupling error.

## Impact
The default behavior is retained. This only adds features.

### PR DESCRIPTION
**Pull request by liamncarlson**
Closes #21434

### CODE REVIEW

File: framework/include/interfaces/Coupleable.h [hunk 1]
```diff
@@ -1170,8 +1170,11 @@ class Coupleable
   /// True if implicit value is required
   bool _c_is_implicit;
 
-  /// Thread ID of the thread using this object
-  THREAD_ID _c_tid;
+  // Argument to allow element-to-nodal coupling
+  bool const _c_allow_element_to_nodal_coupling
```
Comment (GiudGiud) [inline framework/include/interfaces/Coupleable.h:1174]:
```suggestion
  const bool _c_allow_element_to_nodal_coupling;
```

**Review by GiudGiud — CHANGES_REQUESTED** [review]

**Comment by loganharbour** [comment]
@liamncarlson you've still got some clang-formatting to take care of. come chat with me today and we can talk about docuemntation

**Comment by YaqiWang** [comment]
You need to squash all commits into one. You can do
```
git reset HEAD~6
```
first and re-add the changes and commit.

**Review by liamncarlson — COMMENTED** [review]
Changes added

**Review by GiudGiud — APPROVED** [review]

**Pull request merged**